### PR TITLE
[FLINK-29256] Add docs for k8s operator webhook

### PR DESCRIPTION
**Pull request by gaborgsomogyi**
## What is the purpose of the change

Added operator webhook documentation on the architecture page.

## Brief change log

Added operator webhook documentation.

## Verifying this change

Manual:
* `docker run -v $(pwd):/src -p 1313:1313 jakejarvis/hugo-extended:latest server --buildDrafts --buildFuture --bind 0.0.0.0`
* Open in browser: http://localhost:1313/flink/flink-kubernetes-operator-docs-main/docs/concepts/architecture/

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? docs

### CODE REVIEW

File: docs/content/docs/concepts/architecture.md [hunk 1]
```diff
@@ -63,3 +63,18 @@ The Operator manages the lifecycle of Flink resources. The following chart illus
...
   - FAILED : The job terminally failed
+
+## Admission Control
+
+In addition to compiled-in admission plugins, a custom admission plugin named Flink Kubernetes Operator Webhook (Webhook)
+can be started as extension and run as webhook.
+
+The Webhook follow the Kubernetes principles, notably the [dynamic admission control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+It's deployed by default when the Operator installed on a Kubernetes cluster using [Helm](https://helm.sh).
```
Comment (mbalassi) [inline docs/content/docs/concepts/architecture.md:74]: nit: when the operator _is_ installed

**Review by mbalassi — APPROVED** [review]
LGTM, apart from the single typo I marked. Let me know if you have the time to correct otherwise I will when merging @gaborgsomogyi.

File: docs/content/docs/concepts/architecture.md (hunk 1 above)
Comment (gaborgsomogyi) [reply docs/content/docs/concepts/architecture.md:74]: Thanks for the review, fixing it today at some point.

Comment (gaborgsomogyi) [reply docs/content/docs/concepts/architecture.md]: Fixed.

**Pull request merged**